sycl/source/detail/event_impl.cpp

```diff
@@ -159,23 +159,24 @@ event_impl::event_impl(ur_event_handle_t Event, const context &SyclContext)
   }
 }
 
-event_impl::event_impl(const QueueImplPtr &Queue)
-    : MQueue{Queue}, MIsProfilingEnabled{!Queue || Queue->MIsProfilingEnabled} {
-  if (Queue)
-    this->setContextImpl(Queue->getContextImplPtr());
-  else {
-    MState.store(HES_NotComplete);
-    MHostProfilingInfo.reset(new HostProfilingInfo());
-    if (!MHostProfilingInfo)
-      throw sycl::exception(
-          sycl::make_error_code(sycl::errc::runtime),
-          "Out of host memory " +
-              codeToString(UR_RESULT_ERROR_OUT_OF_HOST_MEMORY));
-    return;
-  }
+event_impl::event_impl(queue_impl &Queue)
+    : MQueue{Queue.weak_from_this()},
+      MIsProfilingEnabled{Queue.MIsProfilingEnabled} {
+  this->setContextImpl(Queue.getContextImplPtr());
   MState.store(HES_Complete);
 }
 
+// TODO: comment about https://github.com/intel/llvm/pull/14370
+event_impl::event_impl(std::nullptr_t) : MQueue{}, MIsProfilingEnabled{true} {
```

Comment:
@sergey-semenov , @KseniyaTikhomirova any idea/preferences on how I should change the ctors here?

I think it's very surprising that (before my changes) 
`event_impl e()` and `event_impl e(nullptr)` behave/are implemented so differently.